### ISSUE TITLE
False positive for non standard selectors in no-duplicate-selectors

### ISSUE THREAD

**Issue by cookch10** [stylelint/stylelint#2199]
When using stylelint `v7.6.0` as follows: 

package.json (simplified):

``` json
{
  "scripts": {
    "lint:styles": "stylelint --syntax scss \"src/**/*.scss\""
  }
}
```

My .stylelintrc config (uses stylelint-config-standard `v11.0.0`):

``` json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "function-url-quotes": "none",
    "selector-pseudo-element-colon-notation": "single"
  }
}
```

given the following Sass (scss file): [Sassmeister gist](http://www.sassmeister.com/gist/3f1e0e4872439b81ed369b8d972506ff)

``` scss
.widget-container {
  $variation-without-image: #{"&[data-variation^=without_image]"};
  $has-buttons: #{"&[data-has-buttons]"};

  #{$variation-without-image} {
    & {
      #{$has-buttons} {
        .grid.widget__grid {
          border: 1px solid red;
        }
      }
    }

    .grid.widget__grid {
      border: 1px solid orange;
    }
  }
}
```

``` shell
npm run lint:styles
```

The following results in an error:

``` shell
:5  ×  Unexpected duplicate selector ".widget-container #{$variation-without-image}"   no-duplicate-selectors
```

There is no duplicate selector here.  If I change the Sass to the following (which yields the same output):  given the following Sass (scss file), there is no error observed: [Sassmeister gist](http://www.sassmeister.com/gist/ea0e16f3b14eaa86f7f2ba7355c0bbf2)
``` scss
.widget-container {
  $variation-without-image: #{"&[data-variation^=without_image]"};
  $has-buttons: #{"&[data-has-buttons]"};

  #{$variation-without-image} {
    #{"#{$has-buttons}"} {
      .grid.widget__grid {
        border: 1px solid red;
      }
    }

    .grid.widget__grid {
      border: 1px solid orange;
    }
  }
}
```

**Comment by jeddy3** [comment]
> `:5  ×  Unexpected duplicate selector ".widget-container #{$variation-without-image}"   no-duplicate-selectors`

I think we should ignore any resolved non-standard selectors in this rule i.e. those with interpolation.

**Comment by cookch10** [comment]
@jeddy3 , what are your thoughts on this one?  Is this a valid bug?

**Comment by jeddy3** [comment]
Whoops, I didn't notice those thumbs up.

@cookch10 Yes, this is a bug. Non-standard constructs should be ignored. Would you like to [contribute a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule)? It will involve checking that each _resolved_ selector is [of standard syntax](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxSelector.js). FYI, [tests are run as so](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#running-tests).

**Comment by cookch10** [comment]
@jeddy3, I would be interested in contributing a fix for this.  I'll need a little time to get familiar with the project, so no promises on it being a quick turnaround :)

**Comment by jeddy3** [comment]
> I would be interested in contributing a fix for this.

Awesome! Take your time, there's no rush on these things :)

**Comment by hudochenkov** [comment]
This issue is similar to https://github.com/stylelint/stylelint/issues/2198. The rule works as expected. Explanation why in https://github.com/stylelint/stylelint/issues/2198#issuecomment-274276106.

TL;DR: A rule with a parent selector `&` has the same selector as its ancestor. That is why the rule shows a warning.

```scss
.widget-container {
  $variation-without-image: #{"&[data-variation^=without_image]"};

  #{$variation-without-image} { // ← .widget-container #{$variation-without-image}
    & { // ← .widget-container #{$variation-without-image}
    }
  }
}
```
